File: fixtures/scale/lua/large/checks/check_lcgstream.lua
local lcgstream = require("ciphers.lcgstream")
local lcgstream_spec = require("specs.lcgstream_spec")
local corpus_stream = require("corpus.corpus_stream")

local function check_lcgstream()
  local spec = lcgstream_spec.get()
  for _, text in ipairs(corpus_stream.corpus_stream()) do
    local sealed = lcgstream.lcgstream_encrypt(text, spec.key)
    local opened = lcgstream.lcgstream_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "stream"
end

return { check_lcgstream = check_lcgstream }

File: fixtures/scale/javascript/large/src/checks/checkLcgstream.js
import lcgstream from '../ciphers/lcgstream.js';
import lcgstreamSpecMod from '../specs/lcgstreamSpec.js';
import corpusStreamMod from '../corpus/corpusStream.js';

function checkLcgstream() {
  const spec = lcgstreamSpecMod.lcgstreamSpec();
  for (const text of corpusStreamMod.corpusStream()) {
    const sealed = lcgstream.lcgstreamEncrypt(text, spec.key);
    const opened = lcgstream.lcgstreamDecrypt(sealed, spec.key);
    if (opened !== text) return false;
  }
  return spec.category === 'stream';
}

export default { checkLcgstream };
